### DISCUSSION TITLE
Best practice for accessing object instances across pages

### DISCUSSION THREAD

**Discussion by NanoscaleSimulations** [reflex-dev/reflex#2346] [Q&A]
Hello there,

I am new to web development. I am creating a type of calculator app.
When the user starts a calculation the application creates an instance of a calculator class and has to go through a series of pages where they enter inputs before running a calculation.

My question:
What is the best practice for having access to this calculator instance across pages without having all code in one singe module?

**Comment by masenf** [comment]
I would probably just define the calculator state in its own python module, then import that into each page module that needs to access it. The state itself is shared across the entire app already, so it's just a matter of how you want to structure the python code.

**Reply by NanoscaleSimulations** [reply]
Excellent! 
I misunderstood the concept of states. I thought I had to create an instance of the state and parse it around to work on the same state.

Thank you, I appreciate it :)
